x/deployment/client/cli/tx.go

```diff
@@ -32,10 +40,154 @@ func GetTxCmd(key string, cdc *codec.Codec) *cobra.Command {
 		cmdUpdate(key, cdc),
 		cmdClose(key, cdc),
 		cmdGroupClose(key, cdc),
+		makeItSoCommand(key, cdc),
 	)...)
 	return cmd
 }
 
+func makeItSoCommand(key string, cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "make-it-so [sdl-file]",
+		Short: fmt.Sprintf("Create %s", key),
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Setup CLI context and transaction builder
+			ctx := context.NewCLIContext().WithCodec(cdc)
+			bldr := auth.NewTxBuilderFromCLI(os.Stdin).WithTxEncoder(utils.GetTxEncoder(cdc))
+			if err := ctx.Client.Start(); err != nil {
+				return err
+			}
+
+			// Read in manifest file
+			sdl, err := sdl.ReadFile(args[0])
+			if err != nil {
+				return err
+			}
+
+			// Fetch deployment groups from the SDL file, each group maps to an individual order
+			groups, err := sdl.DeploymentGroups()
+			if err != nil {
+				return err
+			}
+
+			// Fetch the manifest from the SDL file
+			mani, err := sdl.Manifest()
+			if err != nil {
+				return err
+			}
+
+			// Get the deployment ID (defaults to height)
+			id, err := DeploymentIDFromFlags(cmd.Flags(), ctx.GetFromAddress().String())
+			if err != nil {
+				return err
+			}
+
+			// Default DSeq to the current block height
+			if id.DSeq == 0 {
+				if id.DSeq, err = currentBlockHeight(ctx); err != nil {
+					return err
+				}
+			}
+
+			// create a new pubsub bus to handle chain events
+			bus := pubsub.NewBus()
+			defer bus.Close()
+
+			// Create an error group to handle chain listener
+			group, cctx := errgroup.WithContext(ccontext.Background())
+			cctx, cancel := ccontext.WithCancel(cctx)
+
+			// Listen to the events on chain and publish them to the pubsub bus
+			group.Go(func() error {
+				return events.Publish(cctx, ctx.Client, "deployment-create", bus)
+			})
+
+			// expose event channel
+			sub, err := bus.Subscribe()
+			if err != nil {
+				return err
+			}
+
+			// start goroutine to listen for events
+			leases := len(groups)
+			group.Go(func() error {
+				for {
+					select {
+					case <-sub.Done():
+						return nil
+					case ev := <-sub.Events():
+						switch msg := ev.(type) {
+						case types.EventDeploymentCreated:
+							fmt.Printf("Deployment %d created...\n", msg.ID.DSeq)
```

Comment:
Could we use the ctx logger instead of fmt.Print?

Reply:
Thats a much better idea.

Reply:
So turns out you can't get a logger from the `ctx` 🤦

Reply:
the sdk client context - `ctx.PrintOutput()` maybe - it formats in json or yaml depending on flags or whatever.

Reply:
Yeah I want this to look more "log"-y. Think I'll emulate whats in the provider daemon.